src/legacy/core_plugins/elasticsearch/lib/version_health_check.js

```diff
@@ -17,11 +17,23 @@
  * under the License.
  */
 
-import { version as kibanaVersion } from '../../../../../package.json';
+export const versionHealthCheck = (esPlugin, logWithMetadata, esNodesCompatibility$) => {
+  esPlugin.status.yellow('Waiting for Elasticsearch');
 
-export default {
-  // Make the version stubbable to improve testability.
-  get() {
-    return kibanaVersion;
-  },
+  return new Promise(resolve => {
+    esNodesCompatibility$.subscribe(({ isCompatible, message, kibanaVersion, warningNodes }) => {
+      if (!isCompatible) {
+        esPlugin.status.red(message);
+      } else {
+        if (message && message.length > 0) {
+          logWithMetadata(['warning'], message, {
+            kibanaVersion,
+            nodes: warningNodes,
+          });
+        }
+        esPlugin.status.green('Ready');
+        resolve();
+      }
```

Comment:
Should we unsubscribe after the first `resolve` call to avoid wrongly recalling `resolve` in case of green->red->green ?

Reply:
We want to keep updating the status so we need the subscription. Although `resolve()` should only be called once, calling it multiple times is a no-op so it won't cause any problems.